R/zchunk_L221.en_supply.R

```diff
@@ -92,295 +105,661 @@ module_energy_L221.en_supply_DISABLED <- function(command, ...) {
+
+    L111.Prod_EJ_R_F_Yh %>%
+      filter(grepl("unconventional", fuel)) %>%
+      filter(year %in% MODEL_YEARS) %>%
+      left_join(A_regions %>%
+                  select(GCAM_region_ID, region), by = c("GCAM_region_ID")) %>%
+      select(GCAM_region_ID, value, year, region) -> L221.Prod_EJ_R_unoil_Yh
+
+    # Calibrated production of unconventional oil
+    L221.TechCoef_en_Traded %>%
+      filter(supplysector == "traded unconventional oil" & year %in% HISTORICAL_YEARS) %>%
+      left_join(L221.Prod_EJ_R_unoil_Yh %>%
+                  rename(market.name = region), by = c("market.name", "year")) %>%
+      mutate(calOutputValue = round(value, energy.DIGITS_CALOUTPUT)) %>%
+      select(LEVEL2_DATA_NAMES[["TechYr"]], calOutputValue) %>%
+      mutate(calOutputValue = if_else(is.na(calOutputValue), 0, calOutputValue)) %>%
+      mutate(year.share.weight = year) %>%
+      mutate(subsector.share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      mutate(share.weight = if_else(calOutputValue > 0, 1, 0)) -> L221.Production_unoil
+
+    # Unconventional oil demand
+    L121.in_EJ_R_TPES_unoil_Yh %>%
+      left_join(A_regions %>%
+                  select(GCAM_region_ID, region), by = "GCAM_region_ID") -> L121.in_EJ_R_TPES_unoil_Yh
+
+    # Calibrated demand of unconventional oil
+    L221.StubTech_en %>%
+      filter(supplysector == "regional oil" & subsector == "unconventional oil") %>%
+      repeat_add_columns(tibble(year = HISTORICAL_YEARS)) %>%
+      left_join(L121.in_EJ_R_TPES_unoil_Yh, by = c("region", "year")) %>%
+      mutate(calOutputValue = round(value, energy.DIGITS_CALOUTPUT)) %>%
+      mutate(calOutputValue = if_else(is.na(calOutputValue), 0, calOutputValue)) %>%
+      mutate(year.share.weight = year) %>%
+      mutate(subsector.share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      mutate(share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      filter(year %in% MODEL_YEARS) %>%
+      select(region, supplysector, subsector, stub.technology, year, calOutputValue, year.share.weight, subsector.share.weight, share.weight) -> L221.StubTechProd_oil_unoil
+
+    # Crude oil demand
+    L121.in_EJ_R_TPES_crude_Yh %>%
+      left_join(A_regions %>%
+                  select(GCAM_region_ID, region), by = "GCAM_region_ID") -> L121.in_EJ_R_TPES_crude_Yh
+
+    # Calibrated demand of crude oil
+    L221.StubTech_en %>%
+      filter(supplysector == "regional oil" & subsector == "crude oil") %>%
+      repeat_add_columns(tibble(year = HISTORICAL_YEARS)) %>%
+      left_join(L121.in_EJ_R_TPES_crude_Yh, by = c("region", "year")) %>%
+      mutate(calOutputValue = round(value, energy.DIGITS_CALOUTPUT)) %>%
+      mutate(calOutputValue = if_else(is.na(calOutputValue), 0, calOutputValue)) %>%
+      mutate(year.share.weight = year) %>%
+      mutate(subsector.share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      mutate(share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
```

Comment:
Here too pls

Reply:
It looks like exact same logic is happening 3x at least? May want to put into small helper function.